web/app/js/components/TapQueryCliCmd.jsx

```diff
@@ -52,9 +52,9 @@ export default class TapQueryCliCmd extends React.Component {
             <div>Current {_.startCase(this.props.cmdName)} query:</div>
             <code>
               linkerd {this.props.cmdName} {resource}
-              { this.renderCliItem("--namespace", namespace) }
+              { resource.indexOf("namespace") === 0 ? null : this.renderCliItem("--namespace", namespace) }
```

Comment:
I run into a what seems to be a state management issue when messing around with the input fields. When I select a namespace, then remove it and then try to choose a resource from the resource dropdown, the resource list does not show up. It also looks like if I wait long enough, the stats endpoint poll will repopulate the list.

![tap-resource](https://user-images.githubusercontent.com/2197104/44867740-8e965700-ac3d-11e8-8895-2465a351e64f.gif)

Reply:
This particular null check isn't here because we anticipate namespace being null. It's here to allow us to display the CLI command as
```
linkerd tap namespace/emojivoto --max-rps 1.0
```
which is allowed by our CLI, as opposed to 
```
linkerd tap namespace --namespace emojivoto --max-rps 1.0
```
which we do allow, but actually (incorrectly) just taps all namespaces.

we should probably fix the CLI to correctly interpret the latter (a la #1415)

Reply:
It's pretty easy to disable the Resource dropdown if a namespace hasn't been selected, so I can do that!